src/qt/bitcoin.cpp

```diff
@@ -630,7 +630,7 @@ int GuiMain(int argc, char* argv[])
     app.installEventFilter(new GUIUtil::LabelOutOfFocusEventFilter(&app));
 #if defined(Q_OS_WIN)
     // Install global event filter for processing Windows session related Windows messages (WM_QUERYENDSESSION and WM_ENDSESSION)
-    qApp->installNativeEventFilter(new WinShutdownMonitor());
```

Comment:
I'm still not sure if this is safe. The passed in node's shutdown is still a nullptr at this point, because `app.baseInitialize()` has not been called yet, which would in turn call the required initialization functions. I'll have to requisition a windows machine to test this out though.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/28051#discussion_r1304008075

> I'm still not sure if this is safe.

Good catch, It's not safe. The initialization order was fixed by the rewrite https://github.com/bitcoin/bitcoin/pull/28051#issuecomment-1686998476 but then got broken again by the tsan update: https://github.com/bitcoin/bitcoin/pull/28051#issuecomment-1690330644. Should be fixed again now.